### PR TITLE
consensus nodes: add a section on statesyncing

### PR DESCRIPTION
**Pull request by cmwaters**

### CODE REVIEW

File: docs/nodes/consensus-full-node.md
Comment (rootulp) [inline docs/nodes/consensus-full-node.md:107]: [nit] For consistency with https://docs.cometbft.com/v0.34/core/state-sync

```suggestion
#### State sync
```

File: docs/nodes/consensus-full-node.md
Comment (rootulp) [inline docs/nodes/consensus-full-node.md:101]: [question] It looks like block sync is called fast sync on Comet BFT v0.34. Should we use the term "fast sync" and link to https://docs.cometbft.com/v0.34/core/fast-sync here until we upgrade celestia-core to Comet BFT v0.37 which renames fast sync => block sync?

File: docs/nodes/consensus-full-node.md
Comment (rootulp) [reply docs/nodes/consensus-full-node.md:131]: [question] Is Quick sync undocumented in CometBFT docs? I can't find it in https://docs.cometbft.com/v0.34/core/

File: docs/nodes/validator-node.md
Comment (rootulp) [inline docs/nodes/validator-node.md:99]:
```suggestion
State sync uses light client verification to verify state snapshots from peers
```

File: docs/nodes/validator-node.md
Comment (rootulp) [inline docs/nodes/validator-node.md:121]:
```suggestion
#### Quick sync
```

File: docs/nodes/validator-node.md
Comment (rootulp) [inline docs/nodes/validator-node.md:123]:
```suggestion
Quick sync effectively downloads the entire `data` directory from a third-party provider
```

**Review by rootulp — APPROVED** [review]
No blocking feedback. Thanks for adding these sections to docs!

File: docs/nodes/consensus-full-node.md
Comment (cmwaters) [reply docs/nodes/consensus-full-node.md:131]: Yeah it's not documented because it has nothing to do with CometBFT - it's a third party service

File: docs/nodes/consensus-full-node.md
Comment (jcstein) [reply docs/nodes/consensus-full-node.md:101]: I'll defer to @cmwaters on this, and plan to approve this PR after we resolve this 🚢

Comment (jcstein) [reply docs/nodes/consensus-full-node.md:101]: sgtm, lgtm, thanks again

**Pull request merged**
